lib/cfa/placer.rb

```diff
@@ -86,13 +94,16 @@ def initialize(matcher)
 
     # (see Placer#new_element)
     def new_element(tree)
-      index = tree.data.index(&@matcher)
-      res = {}
+      index = tree.data(filtered: false).index(&@matcher)
+      res = create_element
 
       if index
-        tree.data[index] = res
+        # remove old one and add new one, as it can have different key
+        # which cause problem to simple modify
```

Comment:
it was not there before, because we construct it from scratch. But now we only set old key to new value, but it is problem if also key changed. Testcase is in fact there ( see test which use replace placer ). So problem is that with aug.set with new key it will add it to the end. So in fact we need to remove old one from tree and add new one instead of it.